code/game/objects/items/devices/atomic_disassembler.dm

```diff
@@ -0,0 +1,156 @@
+/obj/item/device/atomic_disassembler
+	name = "P.A.D.D."
+	desc = "The Portable Atomic Disassembly Device (patent pending) is a revolutionary device in breaking down objects at the atomic level. To use, you must attach a condensed matter canister and a power \
+	cell. Then, simply feed an object through its atomization field. It will then be broken down into raw condensed matter and placed into the canister, which can be utilized in many ways!"
+	hitsound = 'sound/weapons/genhit1.ogg'
+	w_class = 3 //Maybe make bulky sometime?
+	force = 4
+	slot_flags = SLOT_BACK
+	origin_tech = "programming=4;materials=4;magnets=4;bluespace=4"
+	throw_range = 3
+	throw_speed = 1
+	icon_state = "atom_inactive"
+	mouse_opacity = 2
+	var/obj/item/weapon/stock_parts/cell/power_cell = null
+	var/obj/item/device/compressed_matter_canister/canister = null
+	var/emagged = 0
+	var/list/blacklisted_items = list() //Things that would be considered 'bad' in the lore
+	var/list/really_blacklisted_items = list() //Things that actually break the game
+
+/obj/item/device/atomic_disassembler/New()
+	..()
+	src.assign_items()
+
+/obj/item/device/atomic_disassembler/emag_act(mob/user)
+	..()
+	if(!emagged)
+		user << "<span class='warning'>You disable the safety restrictions on \the [src]!</span>"
+		force = 20 //Disassembling them... ATOMICALLY!
+		damtype = "clone"
+		hitsound = 'sound/weapons/blade1.ogg'
+		emagged = 1
+	return
+
+/obj/item/device/atomic_disassembler/examine(mob/user)
+	..()
+	if(src.emagged)
+		user << "<span class='danger'>The atomization field is flickering randomly.</span>"
+	if(canister)
+		if(canister.stored_matter < canister.matter_cap)
+			user << "<span class='notice'>A small digital display on the canister reads: \"[canister.stored_matter]UCM\".</span>"
+		else
+			user << "<span class='notice'>A small digital display on the canister reads: \"[canister.stored_matter]UCM - CAPACITY REACHED\".</span>"
+
+/obj/item/device/atomic_disassembler/proc/assign_items()
+	blacklisted_items = list(/obj/item/weapon/stock_parts/cell)
+	really_blacklisted_items = list(/obj/item/weapon/reagent_containers/glass, \
+									/obj/item/weapon/disk/nuclear)
+	return 1
```

Comment:
You don't need a proc to do this, just set the definitions above to this exactly as you have them.